www/webfilter/src/opnsense/mvc/app/views/OPNsense/WebFilter/index.volt

```diff
@@ -0,0 +1,268 @@
+
+        // load initial data
+        mapDataToFormUI(data_get_map).done(function(){
+            formatTokenizersUI();
+            $('.selectpicker').selectpicker('refresh');
+            // request service status on load and update status box
+            ajaxCall(url="/api/webfilter/service/status", sendData={}, callback=function(data,status) {
+                updateServiceStatusUI(data['status']);
+            });
+        });
+
+        /***********************************************************************
+         * link grid rules
+         **********************************************************************/
+
+        $("#grid-rules").UIBootgrid(
+            {   search:'/api/webfilter/settings/searchRules',
+                get:'/api/webfilter/settings/getRule/',
+                set:'/api/webfilter/settings/setRule/',
+                add:'/api/webfilter/settings/addRule/',
+                del:'/api/webfilter/settings/delRule/',
+                options: {
+                    rowCount:[10,25,50,100]
+                }
+            }
+        );
+
+        // hook into on-show event for dialog to extend layout.
+        $('#DialogRule').on('shown.bs.modal', function (e) {
+            $("#action\\.type").change(function(){
+                var service_id = 'table_' + $(this).val();
+                $(".type_table").hide();
+                // $(".table_"+$(this).val()).show();
+                $("."+service_id).show();
+            });
+            $("#action\\.type").change();
+        })
+
+        /***********************************************************************
+         * Commands
+         **********************************************************************/
+
+    ajaxCall(url="/api/webfilter/service/download", sendData={}, callback=function(data,status) { 
+        if (data['status'] != 'done') {
+            if (data['status'] == 'running') {
+                $("#dl_categ_progress").addClass("fa fa-spinner fa-pulse");
+                $("#dl_categ").prop("disabled", true);
+                setTimeout(timeoutCheck, 2500);
+            }
+            $('#dl_categ_alert').show();
+        }
+    });
+
+        // Reconfigure webfilter - activate changes
+        $('[id*="reconfigureAct"]').each(function(){
+            $(this).click(function(){
+                // set progress animation
+                $('[id*="reconfigureAct_progress"]').each(function(){
+                    $(this).addClass("fa fa-spinner fa-pulse");
+                });
+
+            });
+        });
+        
+        // form save event handlers for all defined forms
+        $('[id*="save_"]').each(function(){
+            $(this).click(function(){
+                var frm_id = $(this).closest("form").attr("id");
+                var frm_title = $(this).closest("form").attr("data-title");
+
+                // set progress animation
+                $("#"+frm_id+"_progress").addClass("fa fa-spinner fa-pulse");
```

Comment:
If you store that as a variable, you don't have to search it in line 178 again (performance)

Reply:
I used this code from another plugins. I'm not a java guy... help from someone with a better javascript knowledge will be appreciated!

Reply:
you can store the result of the jQuery selector in a local variable:

for example:
```javascript
let progress_icon = $("#"+frm_id+"_progress");
progress_icon.addClass("fa fa-spinner fa-pulse");
```

Then you can use `progress_icon` on the other place as well so you do not have to look it up again and build a jQuery object from it.